css/index.js

```diff
@@ -36,215 +36,240 @@ module.exports = {
-      'border-image-repeat',
-      'outline',
-      'outline-width',
-      'outline-style',
-      'outline-color',
-      'outline-offset',
-      'box-shadow',
-      'opacity',
-      '-ms-interpolation-mode',
-      'page-break-after',
-      'page-break-before',
-      'page-break-inside',
-      'transition',
-      'transition-delay',
-      'transition-timing-function',
-      'transition-duration',
-      'transition-property',
-      'transform',
-      'transform-origin',
-      'perspective',
-      'appearance',
-      'animation',
-      'animation-name',
-      'animation-duration',
-      'animation-play-state',
-      'animation-timing-function',
-      'animation-delay',
-      'animation-iteration-count',
-      'animation-direction',
-      'animation-fill-mode',
-      'fill',
-      'stroke'
+      {
+        groupName: 'Visual',
+        properties: [
+          'cursor',
+          'visibility',
+          'zoom',
+          'table-layout',
+          'empty-cells',
+          'caption-side',
+          'border-spacing',
+          'border-collapse',
+          'content',
+          'quotes',
+          'counter-reset',
+          'counter-increment',
+          'resize',
+          'user-select',
+          'nav-index',
+          'nav-up',
+          'nav-right',
+          'nav-down',
+          'nav-left',
+          'background',
+          'background-color',
+          'background-image',
+          'filter',
+          'background-repeat',
+          'background-attachment',
+          'background-position',
+          'background-position-x',
+          'background-position-y',
+          'background-clip',
+          'background-origin',
+          'background-size',
+          'border',
+          'border-color',
+          'border-style',
+          'border-width',
+          'border-top',
+          'border-top-color',
+          'border-top-style',
+          'border-top-width',
+          'border-right',
+          'border-right-color',
+          'border-right-style',
+          'border-right-width',
+          'border-bottom',
+          'border-bottom-color',
+          'border-bottom-style',
+          'border-bottom-width',
+          'border-left',
+          'border-left-color',
+          'border-left-style',
+          'border-left-width',
+          'border-radius',
+          'border-top-left-radius',
+          'border-top-right-radius',
+          'border-bottom-right-radius',
+          'border-bottom-left-radius',
+          'border-image',
+          'border-image-source',
+          'border-image-slice',
+          'border-image-width',
+          'border-image-outset',
+          'border-image-repeat',
+          'outline',
+          'outline-width',
+          'outline-style',
+          'outline-color',
+          'outline-offset',
+          'box-shadow',
+          'opacity',
+          '-ms-interpolation-mode',
+        ]
+      },
+      {
+        groupName: 'Misc',
```

Comment:
I think even the misc. rules can be split into groups. Over the years (and like many other developers), I've also ended up with an ordering convention similar to @mdo's. But I've never been a fan of group names like "CSS3", "Misc", or "Other", as it suggests something can be improved.

Property groups I currently use are:

1. All properties (`all`)
2. Content (`content`)
3. Position
4. General appearance (`appearance`)
4. Box model
5. Typography
6. Decorations
7. Effects and transforms
8. Interactions
9. Transitions and animations

IMO, it:

- better reflects what's important for rendering the element, putting `content`, `all` or `appearance` up in the list,
- groups "everything"* so there is no "misc"-like group.

\* it's hard to keep the list complete

What can be improved:

- consider reducing/merging single-property groups (1, 2, 4) — these are really special and essential rules, but maybe not worth having their own groups;
- add a group with print related properties like `page-break-after` etc.

What do you guys think?

Reply:
BTW, this is interesting:

> A Stylelint config that sorts CSS properties the way Recess did and Bootstrap did/does.

https://github.com/stormwarning/stylelint-config-recess-order